### PR TITLE
Bug 1857179 - Improved ansible fix for banner files. Replace files on…

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

This patch fix the Bug 1857179 - https://bugzilla.redhat.com/show_bug.cgi?id=1857179
The files / etc / issue and / etc / motd are replaced by the Ansible playbook, even if they are already correct, impacting the efficiency and integrity of the files once a new inode is created.
This patch improves the playbook using the copy module, which ensures that the file is only updated if the current content is different from the desired content.

#### Rationale:

This patch achieve the same goal with less code and more efficiency.
The copy module is part of ansible-base and included in all Ansible installations.

### CODE REVIEW

File: linux_os/guide/system/accounts/accounts-banners/banner_etc_motd/ansible/shared.yml [hunk 1]
```diff
@@ -5,13 +5,7 @@
...
-    path: /etc/motd
-
-- name: "{{{ rule_title }}} - add correct banner"
-  lineinfile:
-    dest: /etc/motd
-    line: '{{{ ansible_deregexify_banner_etc_issue("login_banner_text") }}}'
-    create: yes
+- name: "{{{ rule_title }}}- ensure correct banner"
+  copy:
+    dest: /etc/issue
```
Comment (ggbecker) [inline linux_os/guide/system/accounts/accounts-banners/banner_etc_motd/ansible/shared.yml:10]:
```suggestion
    dest: /etc/motd
```

File: linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/ansible/shared.yml [hunk 2]
```diff
@@ -5,13 +5,7 @@
...
 {{{ ansible_instantiate_variables("login_banner_text") }}}
 
-- name: "{{{ rule_title }}} - remove incorrect banner"
-  file:
-    state: absent
-    path: /etc/issue
-
-- name: "{{{ rule_title }}} - add correct banner"
-  lineinfile:
+- name: "{{{ rule_title }}}- ensure correct banner"
```
Comment (yuumasato) [inline linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/ansible/shared.yml:8]: I have just one nitpick, add a space before the dash, `-`.

File: linux_os/guide/system/accounts/accounts-banners/banner_etc_motd/ansible/shared.yml [hunk 3]
```diff
@@ -5,13 +5,7 @@
...
 {{{ ansible_instantiate_variables("login_banner_text") }}}
 
-- name: "{{{ rule_title }}} - remove incorrect banner"
-  file:
-    state: absent
-    path: /etc/motd
-
-- name: "{{{ rule_title }}} - add correct banner"
-  lineinfile:
+- name: "{{{ rule_title }}}- ensure correct banner"
```
Comment (yuumasato) [inline linux_os/guide/system/accounts/accounts-banners/banner_etc_motd/ansible/shared.yml:8]: Same comment as above.

**Review by yuumasato — CHANGES_REQUESTED** [review]

File: linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/ansible/shared.yml (hunk 2 above)
Comment (marcusburghardt) [reply linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/ansible/shared.yml]: Done. Thanks for catch! : )

**Comment by jan-cerny** [comment]
@marcusburghardt Please rebase and remove the "merge" commits from this PR.

**Comment by jan-cerny** [comment]
Another good idea is to create a feature branch in your repository instead of using master.

**Review by yuumasato — APPROVED** [review]
Thanks!

**Pull request merged**